src/main/java/org/apache/commons/jxpath/Functions.java

```diff
@@ -43,4 +45,17 @@ public interface Functions {
      * @return Function
      */
     Function getFunction(String namespace, String name, Object[] parameters);
+
+    /**
+     * Returns a Function, if any, for the specified namespace,
+     * name and parameter types if the function is allowed by the filter.
+     * @param namespace ns
+     * @param name function name
+     * @param parameters Object[]
+     * @param filter JXPathFilter
+     * @return Function
+     */
+    default Function getFunction(String namespace, String name, Object[] parameters, JXPathFilter filter) {
```

Comment:
Please add `JXPathFilter` as field to `JXPathContext` with the default `new SystemPropertyJXPathFilter()` value. Also add `getPathFilter` and `setPathFilter` methods to it. 

After that, call this `Functions.getFunction` method from `JXPathContextReferenceImpl` with the field value as filter parameter.

Also implement this method in `FunctionLibrary` and `PackageFunctions`, so that the filter parameter can be used for `MethodFunction::new` and `ClassLoaderUtil.getClass`.

Reply:
This will allow the developer to specify custom filter on context level.